cdap-data-fabric/src/main/java/co/cask/cdap/data2/metadata/store/DefaultMetadataStore.java

```diff
@@ -423,6 +428,16 @@ private void publish(MetadataRecord previous, MetadataRecord additions, Metadata
     MetadataChangeRecord.MetadataDiffRecord diff = new MetadataChangeRecord.MetadataDiffRecord(additions, deletions);
     MetadataChangeRecord changeRecord = new MetadataChangeRecord(previous, diff, System.currentTimeMillis());
     changePublisher.publish(changeRecord);
+
+    publishAudit(previous, additions, deletions);
+  }
+
+  private void publishAudit(MetadataRecord previous, MetadataRecord additions, MetadataRecord deletions) {
+    MetadataPayloadBuilder builder = new MetadataPayloadBuilder();
+    builder.addPrevious(previous);
```

Comment:
Chaining methods makes it difficult to read stack trace. Hence I prefer to not chain methods.